packages/flutter_tools/lib/src/base/file_system.dart

```diff
@@ -23,10 +24,13 @@ FileSystem get fs => context == null ? _kLocalFs : context[FileSystem];
 /// Create the ancestor directories of a file path if they do not already exist.
 void ensureDirectoryExists(String filePath) {
   String dirPath = path.dirname(filePath);
-
   if (fs.isDirectorySync(dirPath))
     return;
-  fs.directory(dirPath).createSync(recursive: true);
+  try {
+    fs.directory(dirPath).createSync(recursive: true);
+  } on FileSystemException {
```

Comment:
You could directly use the OSError... Or maybe just use the error message from the error without wrapping it in our own string...